src/rapidfuzz/process.pyi

```diff
@@ -16,110 +16,103 @@ from typing import (
 from rapidfuzz.fuzz import WRatio, ratio
 
 _StringType = Sequence[Hashable]
-_S1 = TypeVar("_S1")
-_S2 = TypeVar("_S2")
+_StringType1 = TypeVar("_StringType1", bound=Sequence[Hashable])
+_StringType2 = TypeVar("_StringType2", bound=Sequence[Hashable])
 _KeyType = TypeVar("_KeyType")
-_ResultType = int | float
+_ResultType = TypeVar("_ResultType")
 
 @overload
 def extractOne(
-    query: _S1,
-    choices: Mapping[_KeyType, _S2 | None],
+    query: _StringType1 | None,
+    choices: Mapping[_KeyType, _StringType2 | None],
     *,
     scorer: Callable[..., _ResultType] = WRatio,
     processor: Callable[..., _StringType] | None = None,
-    score_cutoff: _ResultType | None = None,
-    score_hint: _ResultType | None = None,
+    score_cutoff: float | None = None,
+    score_hint: float | None = None,
```

Comment:
Why was this changed? As far as I can tell, the previous type hint was correct,

Reply:
It was changed taking into account that score_cutoff isn't used for the sake of returns in functions, Result type however it's used for the return of the scorer, which it's not always a float or int, there's cases in the library in which it returns an actual class, such as partial_ratio_alignment

Reply:
But those are very much incompatible with the all of the functions in the `process` module. A scorer can only ever return `int` or `float`.

Reply:
This should still be reverted, since it's not possible to pass a float if the scorer uses int.